src/xmpp/iq.c

```diff
@@ -308,6 +309,21 @@ iq_disco_info_request(gchar *jid)
 }
 
 void
+iq_disco_info_request_onconnect(gchar *jid)
+{
+    xmpp_ctx_t * const ctx = connection_get_ctx();
+    char *id = create_unique_id("disco_info_onconnect");
```

Comment:
Some of the functional tests are failing because they stub responses based on stanza IDs. Using this function has incremented the number used in the unique IDs, so subsequent verifications in the tests fail as the IDs don't match up.

Since this stanza is only sent once on connect we could just give the ID as `disco_info_onconnect` instead of using the unique ID function.

However, since this is actually a problem with the tests (stabber is still in early development and needs more ways to prime/verify stanzas) and not the code, we could also fix all the IDs used in the tests.

The first option is quicker and won't cause any problems in this particular case... thoughts?